core/src/main/java/com/yahoo/imapnio/async/data/ExtendedModifiedSinceTerm.java

```diff
@@ -0,0 +1,130 @@
+ * ABNF:
+ * search-modsequence  = "MODSEQ" [search-modseq-ext] SP mod-sequence-valzer
+ *
+ * entry-name          = entry-flag-name
+ *
+ * entry-flag-name     = DQUOTE "/flags/" attr-flag DQUOTE
+ *                       ;; Each system or user-defined flag <flag>
+ *                       ;; is mapped to "/flags/<flag>".
+ *                       ;;
+ *                       ;; <entry-flag-name> follows the escape rules
+ *                       ;; used by "quoted" string as described in
+ *                       ;; Section 4.3 of [RFC3501]; e.g., for the
+ *                       ;; flag \Seen, the corresponding <entry-name>
+ *                       ;; is "/flags/\\seen", and for the flag
+ *                       ;; $MDNSent, the corresponding <entry-name>
+ *                       ;; is "/flags/$mdnsent".
+ *
+ * entry-type-resp     = "priv" / "shared"
+ *                       ;; Metadata item type.
+ *
+ * entry-type-req      = entry-type-resp / "all"
+ *                       ;; Perform SEARCH operation on a private
+ *                       ;; metadata item, shared metadata item,
+ *                       ;; or both.
+ *
+ * }
+ * </pre>
+ */
+public final class ExtendedModifiedSinceTerm extends SearchTerm {
+
+    /** The name of the metadata item. */
+    private final Flags entryName;
+
+    /** The type of metadata item. */
+    private final EntryTypeReq entryType;
+
+    /** The given modification sequence. */
+    private final long modSeq;
+
+    /**
+     * Constructor.
+     *
+     * @param modSeq modification sequence number
+     */
+    public ExtendedModifiedSinceTerm(final long modSeq) {
+        this.entryName = null;
+        this.entryType = null;
+        this.modSeq = modSeq;
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param entryName name of the metadata item
+     * @param entryType type of the metadata item
+     * @param modSeq modification sequence number
+     */
+    public ExtendedModifiedSinceTerm(@Nonnull final Flags entryName, @Nonnull final EntryTypeReq entryType, final long modSeq) {
+        this.entryName = entryName;
+        this.entryType = entryType;
+        this.modSeq = modSeq;
+    }
+
+    /**
+     * @return the entry name
+     */
+    @Nullable
+    public Flags getEntryName() {
+        return entryName;
+    }
+
+    /**
+     * @return the entry type
+     */
+    @Nullable
+    public EntryTypeReq getEntryType() {
+        return entryType;
+    }
+
+    /**
+     * @return the modification sequence
+     */
+    public long getModSeq() {
+        return modSeq;
+    }
+
+    /**
+     * The match method.
+     *
+     * @param msg the date comparator is applied to this Message's MODSEQ
+     * @return true if the comparison succeeds, otherwise false
+     */
+    @Override
+    public boolean match(final Message msg) {
+        try {
+            if (msg instanceof IMAPMessage) {
+                final long m = ((IMAPMessage) msg).getModSeq();
```

Comment:
To extend SearchTerm, I have to override match function, but I cannot compare entry name and entry type without extended IMAPMessage. I think I should just return false and make comments above function to avoid users to use this function.